### ISSUE TITLE
Pipfile.lock does not get generated with a newline at the end of the file.

### ISSUE THREAD

**Issue by thomasdziedzic** [kennethreitz/pipenv#319]
I have been adding newlines manually after each generation to workaround this issue.

http://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

```
$ pipenv lock
Locking [dev-packages] dependencies...
Locking [packages] dependencies...
Updated Pipfile.lock!
```

```
$ git diff
diff --git a/Pipfile.lock b/Pipfile.lock
index d9621453..b7e6d6c8 100644
--- a/Pipfile.lock
+++ b/Pipfile.lock
@@ -67,4 +67,4 @@
             "version": "==1.3.7"
         }
     }
-}
+}
\ No newline at end of file
```

**Comment by nateprewitt** [comment]
Hey @thomasdziedzic, thanks for starting this discussion. So the lack of newline is a design decision of the standard libraries `json` module. You can see this behaviour with any value passed into `json.dumps` which is how we're storing this. `Pipfile.lock` isn't intended to be touched by anything except the installation tool (in this case Pipenv). I'm assuming your text editor is opening and auto-saving the Pipfile.lock with a newline? The issue you're encountering appears to be a byproduct of your development environment, rather than something specific to pipenv.

That said, I don't think there's a big reason not to have a newline, other than it *may* affect some parsers. @kennethreitz, do you have thoughts here. This is a pretty trivial fix and doesn't appear to break anything from initial testing.

**Comment by thomasdziedzic** [comment]
> I'm assuming your text editor is opening and auto-saving the Pipfile.lock with a newline?

Just to clarify, I am opening and saving this file explicitly to generate the newline at the end because I want this newline.

**Comment by nateprewitt** [comment]
This is resolved in f5f0f71, thanks again for your input @thomasdziedzic!
